src/main/java/com/google/maps/StaticMapsRequest.java

```diff
@@ -0,0 +1,415 @@
+   *
+   * @param scale The scale of the static map.
+   * @return Returns this {@code StaticMapsRequest} for call chaining.
+   */
+  public StaticMapsRequest scale(int scale) {
+    return param("scale", scale);
+  }
+
+  public enum ImageFormat implements UrlValue {
+    png("png"),
+    png8("png8"),
+    png32("png32"),
+    gif("gif"),
+    jpg("jpg"),
+    jpgBaseline("jpg-baseline");
+
+    private final String format;
+
+    ImageFormat(String format) {
+      this.format = format;
+    }
+
+    @Override
+    public String toUrlValue() {
+      return format;
+    }
+  }
+
+  /**
+   * <code>format</code> defines the format of the resulting image. By default, the Google Static
+   * Maps API creates PNG images. There are several possible formats including GIF, JPEG and PNG
+   * types.
+   *
+   * @param format The format of the static map.
+   * @return Returns this {@code StaticMapsRequest} for call chaining.
+   */
+  public StaticMapsRequest format(ImageFormat format) {
+    return param("format", format);
+  }
+
+  public enum StaticMapType implements UrlValue {
+    roadmap,
+    satellite,
+    terrain,
+    hybrid;
+
+    public String toUrlValue() {
+      return this.name();
+    }
+  }
+
+  /**
+   * <code>maptype</code> defines the type of map to construct.
+   *
+   * @param maptype The map type of the static map.
+   * @return Returns this {@code StaticMapsRequest} for call chaining.
+   */
+  public StaticMapsRequest maptype(StaticMapType maptype) {
+    return param("maptype", maptype);
+  }
+
+  /**
+   * <code>region</code> defines the appropriate borders to display, based on geo-political
+   * sensitivities. Accepts a region code specified as a two-character ccTLD ('top-level domain')
+   * value.
+   *
+   * @param region The region of the static map.
+   * @return Returns this {@code StaticMapsRequest} for call chaining.
+   */
+  public StaticMapsRequest region(String region) {
+    return param("region", region);
+  }
+
+  private static final Pattern colorPattern =
+      Pattern.compile(
+          "(0x\\p{XDigit}{6})|(black)|(brown)|(green)|(purple)|(yellow)|(blue)|(gray)|(orange)|(red)|(white)");
+
+  public static class Markers implements UrlValue {
+
+    public static enum MarkersSize implements UrlValue {
+      tiny,
+      mid,
+      small,
+      normal;
+
+      @Override
+      public String toUrlValue() {
+        return this.name();
+      }
+    }
+
+    public static enum CustomIconAnchor implements UrlValue {
+      top,
+      bottom,
+      left,
+      right,
+      center,
+      topleft,
+      topright,
+      bottomleft,
+      bottomright;
+
+      @Override
+      public String toUrlValue() {
+        return this.name();
+      }
+    }
+
+    private MarkersSize size;
+    private String color;
+    private String label;
+    private String customIconURL;
+    private CustomIconAnchor anchorPoint;
+    private List<String> locations = new ArrayList<>();
```

Comment:
private final, you don't ever change this ref

Reply:
Done